canvas/tree.py

```diff
@@ -0,0 +1,111 @@
+def _is_collapsible(tree):
+    """ Checks to see if the children of the node can be collapsed. """
+    if len(tree.children) == 0:
+        return False
+
+    for c in tree.children:
+        if not c.is_tip():
+            return False
+    return True
+
+
+def _trim_level(tree, table):
+    """ Collapses tree leaves together by one level. """
+    collapsed_tree = tree.copy()
```

Comment:
If we put this into scikit-bio, no.  Since we'll be modifying the Tree object directly.

However, in this API, all of the collapsing will be done with out modifying the original tree.  I wanted to do this, since it isn't clear how to expose the `skbio.TreeNode` api to enable this.

Does that make sense?